_posts/2015-02-23-taskcluster-update.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Taskcluster Release Part 1 : Gecko"
+description: "TC Update 1"
+category: mozilla, ci
+tags: [ci, docker, mozilla]
+---
+
+It's been awhile since my last blog post about taskcluster and I wanted to give an update...
+
+## Taskcluster + Gecko
+
+Taskcluster is running by default on
+
+  - [try](https://treeherder.allizom.org/#/jobs?repo=try)
+  - [b2g-inbound](https://treeherder.allizom.org/#/jobs?repo=b2g-inbound)
+  - [mozilla-inbound](https://treeherder.allizom.org/#/jobs?repo=mozilla-inbound)
+  - [fx-team](https://treeherder.allizom.org/#/jobs?repo=fx-team)
+  - [mozilla-central](https://treeherder.allizom.org/#/jobs?repo=mozilla-central)
+
+
+You will see both buildbot run and taskcluster run jobs the "TC" jobs are
+prefixed accordingly so you can tell the difference.
+
+This is the last big step to enabling TC as the default CI for many mozilla
+projects also new branches are a simple config away.
+
+Why is this a great thing? Just about everything is [in](https://dxr.mozilla.org/mozilla-central/source/testing/taskcluster/mach_commands.py) [the](https://dxr.mozilla.org/mozilla-central/source/testing/docker) [tree](https://dxr.mozilla.org/mozilla-central/source/testing/taskcluster/tasks/builds).
+
+This means you can easily add new builds/tests and immediately push them
+to try for testing (see the configs for [try](https://dxr.mozilla.org/mozilla-central/source/testing/taskcluster/tasks/branches/try/job_flags.yml)
+
+Results currently show up only on [staging treeherder](https://treeherder.allizom.org)
+aside from the ease of adding tests there are a few other key benefits
+on linux:
+
+#### We use [docker](https://dxr.mozilla.org/mozilla-central/source/testing/docker)
+
+Which means a completely identical CI environment is a docker pull away.
+
+```sh
+# Pull tester image
+docker pull quay.io/mozilla/tester:0.0.14
+# Run tester image shell
+docker run -it quay.io/mozilla/tester:0.0.14 /bin/bash
+# <copy/paste stuff from task defintions into this>
+```
+
+#### Generally tests and builds are faster
+
+Through this entire process we have been optimizing away overhead and using faster
+machines which means both build (and particularly test) times are faster.
+
+(Wins look [big](https://treeherder.allizom.org/#/jobs?repo=b2g-inbound&revision=233af1dfa476) but more in future blog post)
+
+#### What's missing ?
+
+There are a few missing pieces:
```

Comment:
We can axe this sentence and simply jump into the list of things we are missing. :)